## Step 1: Import Libraries
Let's start by importing the core data science libraries we will use: NumPy, and Pandas.


In [3]:
import pandas as pd
import numpy as np

---
## Step 2: Load and Inspect the Data
Now we will load the Titanic dataset (`tested.csv`) into a Pandas DataFrame and check its shape and structure.    
Then drop NaN values so they don't intervene with calculating the statistical formulas.

In [4]:
df = pd.read_csv("../../Data/1.TitanicDataset.csv")
print("Dataset Info:")
df.info()
df = df.dropna(subset=["Age"])

Dataset Info:
<class 'pandas.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null    int64  
 1   Survived     418 non-null    int64  
 2   Pclass       418 non-null    int64  
 3   Name         418 non-null    str    
 4   Sex          418 non-null    str    
 5   Age          332 non-null    float64
 6   SibSp        418 non-null    int64  
 7   Parch        418 non-null    int64  
 8   Ticket       418 non-null    str    
 9   Fare         417 non-null    float64
 10  Cabin        91 non-null     str    
 11  Embarked     418 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 39.3 KB


---
## Step 3: Compute the mean, median, mode, standard deviation, and IQR.

In [21]:
Age = df['Age']
print("Mean of the column Age =", np.mean(Age)) 
print("Median of the column Age =", np.median(Age)) 
print("Mode of the column Age =", Age.mode()[0]) 
print("Standard Deviation of the column Age =", np.std(Age))
print("Variance of the column Age =", np.var(Age)) 
q1, q3 = np.percentile(Age, [25, 75])
print("IQR of the column Age =", q3 - q1)


Mean of the column Age = 30.272590361445783
Median of the column Age = 27.0
Mode of the column Age = 21.0
Standard Deviation of the column Age = 14.159835886272559
Variance of the column Age = 200.50095232617218
IQR of the column Age = 18.0



## Step 4:  Identify whether the mean or median better represents a 'typical' value, and justify the choice.
Using the median here is better to represent the typical age of people on the ship; because the bulk of the passengers are of age 18-27 but there's some Older people of Age 60-80 (Outliers) whom skew the dataset to the right 

---
## Step 5 : Write a Markdown cell summarizing the column's center and spread in plain language.
The column's center is represented by the Median at 27.0.                                                                  
the spread is pretty high since the range of Ages on the ship was big (from 0-70 Years old), in numbers the Standard Deviation is 14.15, the variance is 200.5 and the IQR is 18.0.